### DISCUSSION TITLE
SSR Helper mutations

### DISCUSSION THREAD

**Discussion by maritz** [trpc/trpc#4682] [Q&A] [answered]
When using SSR helpers, I don't see a way to use mutations. In the documentation I see this:

> However, rather than useQuery and useMutation, you get prefetch, fetch, prefetchInfinite, and fetchInfinite functions.

But when I try to access these fetch methods, the mutations are not there at all.

Given this: 
```typescript
export const userRouter = createTRPCRouter({
  registerUser: publicProcedure
    .input(registerUserInputValidator)
    .mutation(registerUserMethod),
});

// and then later userRouter is merged into my appRouter
```

And then trying to access it in an SSR helper in getServerSideProps:

```typescript
const helpers = await createSSRHelpers(context, appRouter, session);
const registrationResult = await helpers.user.registerUser.fetch({ some: 'data' });
//                                            ^- Property 'registerUser' does not exist on type 'DecoratedProcedureSSGRecord....
```

I'm not sure if I'm misunderstanding things or whether this is a bug.

As a workaround for now I have duplicated my mutation as a normal query that is only to be used in SSR helpers. But obviously that's not ideal.

**Comment by KATT** [comment, marked as answer]
You can just create a "caller" of your router and use that for mutations: https://trpc.io/docs/server/server-side-calls

**Reply by maritz** [reply]
Right, I had totally forgotten about those. Thank you.

A note in the documentation about mutations not being available in helpers might be useful though, right?
Because as it stands now the quoted text above makes it seem like they should be in my reading of it.
